testcase/helper_test.go

```diff
@@ -6,32 +6,55 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-// TestParseDotProperty test keys that contain dot or bracket notation are converted to sub structure
-func TestParseDotProperty(t *testing.T) {
-
+func TestExtractBracketFields(t *testing.T) {
 	var (
 		key      string
-		value    string
-		result   map[string]interface{}
-		expected map[string]interface{}
+		result   []string
+		expected []string
 	)
 
-	// Create sub structure with dot notation
-	key = "log.file.path"
-	value = "/tmp/test.log"
-	result = make(map[string]interface{})
-	expected = map[string]interface{}{
-		"log": map[string]interface{}{
-			"file": map[string]interface{}{
-				"path": "/tmp/test.log",
-			},
-		},
+	// When bracket
+	key = "[log][file][path]"
+	expected = []string{
+		"log",
+		"file",
+		"path",
+	}
+	result = extractBracketFields(key)
+	assert.Equal(t, expected, result)
+
+	// When no bracket
+	key = "message"
+	expected = []string{
+		"message",
 	}
-	parseDotProperty(key, value, result)
+	result = extractBracketFields(key)
 	assert.Equal(t, expected, result)
 
+	// When bad bracket
+	key = "[log]badformat"
+	expected = []string{
+		"[log]badformat",
+	}
+
```

Comment:
There's no assertion here. Apart from a non-working test the code won't pass the ineffassign linter:

```
$ make check 
golangci-lint run
testcase/helper_test.go:35:2: ineffectual assignment to `key` (ineffassign)
	key = "[log]badformat"
	^
testcase/helper_test.go:36:2: ineffectual assignment to `expected` (ineffassign)
	expected = []string{
	^
Makefile:76: recipe for target 'check' failed
make: *** [check] Error 1
```

Normally this would've been caught by Travis, but it seems that the fact that your reused PR branch conflicts with master causes Travis to not run. To fix this, rebase your top two commits onto the upstream master and force-push to your branch.

```
git fetch
git rebase -i --onto 8b9c241f HEAD~2
git push -f origin master
```

This can be avoided by not starting new PRs from old PR branches.

Reply:
`make check` stills fails for the same reason as earlier. Here's the Travis build that was run but for some reason wasn't posted to GitHub: https://travis-ci.org/magnusbaeck/logstash-filter-verifier/builds/627112921

You got the rebase right but that in itself didn't fix the underlying problem of the bad code.

Reply:
`make check` now work as expected.